app/views/requests/index.html.erb

```diff
@@ -61,6 +61,12 @@
                 <%= clear_filter_button %>
                 <%= modal_button_to("#calculateTotals", {text: "Calculate Product Totals", icon: nil, size: "md", type: "success"}) %>
                 <span class="float-right">
+                  <% if @unfulfilled_requests.exists? %>
```

Comment:
I usually use `any?` here.

Reply:
Did a little deep dive into `any?` vs `exists?`. Learned that since the records are not pre-loaded, when calling `any?` in the index view, it's delegating to `exists?` anyhow and not loading any records. Then a few lines down on line 67, I'm calling `.count` which runs a second query. I realize for the index view, I only need the count and not the loaded records; therefore, I changed the `@unfulfilled_requests` instance variable to `@unfulfilled_requests_count` in the Requests controller.